packages/app-builder/src/routes/ressources+/scenarios+/$scenarioId+/testrun+/$testRunId+/cancel.tsx

```diff
@@ -0,0 +1,101 @@
+import { useFetcher, useNavigation } from '@remix-run/react';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHydrated } from 'remix-utils/use-hydrated';
+import { Button, ModalV2 } from 'ui-design-system';
+
+export async function action({ request, params }: ActionFunctionArgs) {
+  const { authService } = serverServices;
+  const scenarioId = fromParams(params, 'scenarioId');
+  const testRunId = fromParams(params, 'testRunId');
+  const { testRun } = await authService.isAuthenticated(request, {
+    failureRedirect: getRoute('/sign-in'),
+  });
+
+  try {
+    await testRun.cancelTestRun({ testRunId });
+    return redirect(
+      getRoute('/scenarios/:scenarioId/test-run/', {
+        scenarioId: fromUUID(scenarioId),
+      }),
+    );
+  } catch (error) {
+    const { getSession, commitSession } = serverServices.toastSessionService;
+
+    const session = await getSession(request);
+
+    setToastMessage(session, {
+      type: 'error',
+      messageKey: 'common:errors.unknown',
+    });
+
+    return json(
+      { success: false as const },
+      { headers: { 'Set-Cookie': await commitSession(session) } },
+    );
+  }
+}
+
+export function CancelTestRun({
+  children,
+  testRunId,
+}: {
+  children: React.ReactElement;
+  testRunId: string;
+}) {
+  const [open, setOpen] = useState(false);
+  const hydrated = useHydrated();
+  const navigation = useNavigation();
+  const currentScenario = useCurrentScenario();
+  const cancelTestRunFetcher = useFetcher<typeof action>();
+  const { t } = useTranslation(['scenarios', 'common']);
+
+  useEffect(() => {
+    if (navigation.state === 'loading') {
+      setOpen(false);
+    }
+  }, [navigation.state]);
+
+  return (
+    <ModalV2.Root open={open} setOpen={setOpen}>
+      <ModalV2.Trigger render={children} disabled={!hydrated} />
+      <ModalV2.Content className="overflow-visible">
+        <ModalV2.Title>{t('scenarios:testrun.cancel.title')}</ModalV2.Title>
+        <cancelTestRunFetcher.Form
+          className="flex flex-col gap-6 p-6"
+          method="POST"
+          action={getRoute(
+            '/ressources/scenarios/:scenarioId/testrun/:testRunId/cancel',
+            {
+              scenarioId: fromUUID(currentScenario.id),
+              testRunId: fromUUID(testRunId),
+            },
+          )}
+        >
+          <ModalV2.Description render={<Callout variant="outlined" />}>
+            <p className="whitespace-pre text-wrap">
+              {t('scenarios:testrun.cancel.callout')}
+            </p>
+          </ModalV2.Description>
+          <div className="flex flex-1 flex-row gap-2">
+            <ModalV2.Close
+              render={<Button className="flex-1" variant="secondary" />}
+            >
+              {t('common:no')}
+            </ModalV2.Close>
+            <Button className="flex-1" variant="primary" type="submit">
+              {t('common:yes')}
+            </Button>
+          </div>
```

Comment:
**tl;dr**: 
- `"cancel"` is used in all validation modals to let the user revert the change
- `"ACTION_NAME"` is used in the modal trigger, modal title and modal confirmation button

To prevent confusion, we should rename "cancel" testrun for "stop"/"deactivte" (ask in #product chan)

___

I prefer to use "action verb" for validation (so user do not need to read again the modal context to understand what is going on )
```suggestion
          <div className="flex flex-1 flex-row gap-2">
            <ModalV2.Close
              render={<Button className="flex-1" variant="secondary" />}
            >
              {t('common:cancel')}
            </ModalV2.Close>
            <Button className="flex-1" variant="primary" type="submit">
              {t('scenarios:testrun.cancel')}
            </Button>
          </div>
```

And thus, because "cancel" is used for both labels, I would recommand to change "cancel test run" for "stop test run" (and alike, I think you get it)